man/systemd-coredump.xml

```diff
@@ -141,12 +142,233 @@
+COREDUMP_CMDLINE=/usr/lib64/firefox/firefox -contentproc -childID 5 -isForBrowser …
+COREDUMP_CGROUP=/user.slice/user-1000.slice/user@1000.service/app.slice/app-….scope
+COREDUMP_FILENAME=/var/lib/systemd/coredump/core.Web….552351.….zst
+…
+    </programlisting>
+
+    <para>The following fields are saved (if known) with the journal entry</para>
+
+    <variablelist class='journal-directives'>
+      <varlistentry>
+        <term><varname>COREDUMP_UID=</varname></term>
+        <term><varname>COREDUMP_PID=</varname></term>
+        <term><varname>COREDUMP_GID=</varname></term>
+        <listitem><para>The process number (PID) and owner (UID and GID) of the crashed process.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>COREDUMP_TIMESTAMP=</varname></term>
+        <listitem><para>The time of the crash as reported by the kernel (in µs since the epoch).</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>COREDUMP_RLIMIT=</varname></term>
+        <listitem><para>The core file size soft resource limit, see
+        <citerefentry project='man-pages'><refentrytitle>getrlimit</refentrytitle><manvolnum>2</manvolnum></citerefentry>.
+        </para></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>COREDUMP_UNIT=</varname></term>
+        <term><varname>COREDUMP_SLICE=</varname></term>
+        <listitem><para>The system unit and slice names.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>COREDUMP_CGROUP=</varname></term>
+        <listitem><para>Control group information in the format used in
+        <filename>/proc/self/cgroup</filename>. On systems with the unified cgroup hierarchy, this is a
+        single path prefixed with <literal>0::</literal>, and multiple paths prefixed with controller numbers
+        on legacy systems.</para></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>COREDUMP_OWNER_UID=</varname></term>
+        <term><varname>COREDUMP_USER_UNIT=</varname></term>
+        <listitem><para>The numerical UID of the user owning the login session or systemd user unit of the
+        crashed process, and the user manager unit. Both fields are only present for user processes.
+        </para></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>COREDUMP_SIGNAL_NAME=</varname></term>
+        <term><varname>COREDUMP_SIGNAL=</varname></term>
+
+        <listitem><para>The terminating signal name and numerical value. (Both are included because signal
+        numbers vary by architecture.)</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>COREDUMP_CWD=</varname></term>
+        <term><varname>COREDUMP_ROOT=</varname></term>
+
+        <listitem><para>The current working directory and root directory of the crashed process.</para>
```

Comment:
should probably say relative to the host fs hierarchy (and not the container's mount namespace, if there is one)

Reply:
Except it's not... Now I'm confused. All other fields are from the outside, but those paths are relative to container root.

Reply:
hmm, whaaa? i assumed that /proc/$PID/root symlinks would expose paths relative to the caller. super confused now.

Reply:
I have the following example  I was using to test this:
```
    COREDUMP_SIGNAL_NAME=SIGABRT
    COREDUMP_HOSTNAME=bionic
    COREDUMP_COMM=bash
    COREDUMP_EXE=/bin/bash
    COREDUMP_UNIT=machine-bionic.scope
    COREDUMP_SLICE=machine.slice
    COREDUMP_CMDLINE=bash -c kill -ABRT $$
    COREDUMP_CGROUP=/machine.slice/machine-bionic.scope/payload
    COREDUMP_OPEN_FDS=0:/dev/pts/0
          …
    COREDUMP_PROC_CGROUP=0::/machine.slice/machine-bionic.scope/payload
    COREDUMP_CWD=/root
    COREDUMP_ROOT=/
    COREDUMP_PID=866139
    COREDUMP_UID=1449984000
    COREDUMP_GID=1449984000
    COREDUMP_PROC_MAPS=5608fee00000-5608fef04000 r-xp 00000000 fd:01 568339                     /bin/bash
                       5608ff103000-5608ff107000 r--p 00103000 fd:01 568339                     /bin/bash
                       7fcc5acea000-7fcc5acf5000 r-xp 00000000 fd:01 568996                     /lib/x86_64-linux-gnu/libnss…
    COREDUMP_PROC_MOUNTINFO=1235 1200 253:1 /var/lib/machines/bionic / rw,relatime shared:247 master:1 - ext4 /dev/mapper/fe>
                            1236 1235 0:94 / /tmp rw,nosuid,nodev shared:386 - tmpfs tmpfs rw,size=801056k,nr_inodes=409600,>
                            1237 1235 0:22 / /sys ro,nosuid,nodev,noexec,relatime shared:418 - sysfs sysfs rw
                            1238 1235 0:95 / /dev rw,nosuid shared:419 - tmpfs tmpfs rw,size=4096k,nr_inodes=65536,mode=755,…
                            1239 1238 0:96 / /dev/shm rw,nosuid,nodev shared:420 - tmpfs tmpfs rw,size=801056k,nr_inodes=409…
                            1245 1241 253:1 /usr/lib/os-release /run/host/os-release rw,relatime shared:426 master:1 - ext4…
    COREDUMP_CONTAINER_CMDLINE=systemd-nspawn -U -M bionic
    MESSAGE=Process 866139 (bash) of user 1449984000 dumped core.
…
```

So yeah, we're freely mixing namespaced and non-namespaced stuff.
Also, the commandlines are rendered in a way that is less than fully useful.
Plenty of stuff to improve here.